app/api/search/search.js

```diff
@@ -600,40 +512,28 @@ const entityHasGeolocation = entity =>
       return false;
     });
 
-const processGeolocationResults = (_results, templatesInheritedProperties, inheritedEntities) => {
+const processGeolocationResults = _results => {
   const results = _results;
-  const processedRows = [];
-  const affectedTemplates = Object.keys(templatesInheritedProperties);
-
-  results.rows.forEach(_row => {
-    const row = _row;
-    if (affectedTemplates.includes(row.template)) {
-      Object.keys(row.metadata).forEach(property => {
-        if (templatesInheritedProperties[row.template][property]) {
-          row.metadata[property].forEach(({ value: entity }, index) => {
-            const targetProperty = templatesInheritedProperties[row.template][property].target.name;
-            const inherited = inheritedEntities[entity]
-              ? inheritedEntities[entity]
-              : { metadata: {} };
-            inherited.metadata = inherited.metadata || {};
-            row.metadata[property][index] = row.metadata[property][index] || {};
-            row.metadata[property][index] = {
-              ...row.metadata[property][index],
-              value: entity,
-              // geolocation may not be sanitized...
-              inherit_geolocation: (inherited.metadata[targetProperty] || []).filter(p => p.value),
-            };
-          });
-        }
-      });
-    }
-    if (entityHasGeolocation(row)) {
-      processedRows.push(row);
-    }
-  });
-
-  results.rows = processedRows;
-  results.totalRows = processedRows.length;
+  results.rows = results.rows.filter(r => r.metadata);
+  results.rows = results.rows
+    .map(_row => ({
+      ..._row,
+      metadata: Object.fromEntries(
+        Object.entries(_row.metadata).map(([key, values]) => [
+          key,
+          values.map(_v => {
+            const newValue = { ..._v };
+            if (_v.inheritedType === propertyTypes.geolocation) {
+              newValue.inherit_geolocation = (_v.inheritedValue || []).filter(iv => iv.value);
+            }
+            return newValue;
+          }),
+        ])
+      ),
+    }))
+    .filter(r => r.metadata && Object.keys(r.metadata).length);
+  results.rows = results.rows.filter(r => entityHasGeolocation(r));
```

Comment:
In the previous implementation, there was one iteration over rows.  

Here you have your iteration over the entire array in line 517.  Then a map and filter on 518.  Then again a new filter in 534.

I'm wondering if this impacts performance for very large array of results?  Also, can't they be somehow combined?, for example, can't line 534 also include and `&& entityHasGeolocation(r)` instead of having to iterate again over the entire array?